ocaml-lsp-server/src/progress.ml

```diff
@@ -0,0 +1,78 @@
+open Import
+
+type enabled =
+  { mutable token : ProgressToken.t option
+  ; mutable build_counter : int
+  ; progress_report :
+      Server_notification.Progress.t ProgressParams.t -> unit Fiber.t
+  ; create_task : WorkDoneProgressCreateParams.t -> unit Fiber.t
+  }
+
+type t =
+  | Disabled
+  | Enabled of enabled
+
+let create (client_capabilities : ClientCapabilities.t) progress_report
```

Comment:
I would only send the relevant capability here, to reduce the dependencies of the module

Reply:
If the option was always present, I would do that, but I'd rather not split the logic for the default value and the flag itself across functions.